RevenueCatUI/Templates/V2/Components/Packages/PurchaseButton/PurchaseButtonComponentView.swift

```diff
@@ -139,7 +145,13 @@ struct PurchaseButtonComponentView: View {
                            inAppBrowserURL: self.$inAppBrowserURL)
 
         if launchWebCheckout.autoDismiss {
-            self.onDismiss()
+            if case .inAppBrowser = method {
+                self.inAppBrowserDidDisappearCompletion = {
+                    self.onDismiss()
```

Comment:
Hmm I wonder... in this case, would it make sense to do something like:
- Fetching new CustomerInfo from backend (display some loading state in the meantime).
- If it has a new purchase, then dismiss, otherwise keep the paywall displaying.

Mostly thinking that a purchase could have been cancelled here... Not sure if there is some nice way to communicate this info from the web view directly though... Also, this might be acceptable for now though... Just possibly could cause some weird behavior when closing the web

Reply:
Thanks for your suggestion @tonidero!

Exactly! Actually, that is what I meant in [my comment above](https://github.com/RevenueCat/purchases-ios/pull/5114#issuecomment-2890009018). In fact, I did implement something similar in #5083 (now discarded).

Now, I agree that we could be smarter about this with your proposed approach. However, should that be the expected behavior for "Auto dismiss" (i.e. "Auto dismiss == auto dismiss when purchase successfully made")?

As you said, I think we should get this fix out as is (otherwise, in-app browser web checkout would be broken as the the whole paywall would get dismissed when presenting the in-app browser in this case). Then, we can discuss about whether or not we want to be more clever about when to auto dismiss.

cc @joshdholtz

Reply:
Ah right, that makes sense! I'm ok merging this for now yeah and we can modify  later if needed.